docs/content/guides/dagster/transitioning-data-pipelines-from-development-to-production.mdx

```diff
@@ -0,0 +1,456 @@
+
+Now, we can set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<!-- <Note>
+If you're using <a href="https://dagster.io/cloud">Dagster Cloud</a>, this environment variable will be set for you. Additionally, if you're using <a href="https://docs.dagster.cloud/guides/branch-deployments">Branch Deployments</a> <code>DAGSTER_DEPLOYMENT</code> will be automatically set to <code>VAR</code>.
+
+</Note> -->
+
+We still have some problems with this setup:
+
+1. Developers need to remember to change `user` and `password` to their credentials and `schema` to their name when developing locally.
+2. Passwords are being stored in code.
+
+We can easily solve these problems because the Snowflake I/O manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables and point Dagster to those environment variables in the run configuration:
+
+```python file=/guides/dagster/development_to_production/repository/repository_v3.py startafter=start endbefore=end
+# repository.py
+
+
+@repository
+def repo():
+    resource_defs = {
+        "local": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": {"env": "DEV_SNOWFLAKE_USER"},
+                    "password": {"env": "DEV_SNOWFLAKE_PASSWORD"},
+                    "database": "LOCAL",
+                    "schema": {"env": "DEV_SNOWFLAKE_SCHEMA"},
+                }
+            ),
+        },
+        "production": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "system@company.com",
+                    "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS",
+                }
+            ),
+        },
+    }
+    deployment_name = os.getenv("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        *with_resources(
+            [items, comments, stories], resource_defs=resource_defs[deployment_name]
+        )
+    ]
+```
+
+### Staging
+
+Depending on your organization’s Dagster setup, there are a couple of options for a staging environment.
+
+- **For Dagster Cloud users**, we recommend using [Branch Deployments](https://docs.dagster.cloud/guides/branch-deployments) as your staging step. A Branch Deployment is a new Dagster deployment that is automatically generated for each git branch. Check out our \[comprehensive guide to Branch Deployments]\(link to claire’s doc) to learn how to use Branch Deployments to verify data pipelines before deploying them to production.
```

Comment:
nit: should branch deployments be lowercased?

Reply:
idk @erinkcochran87 ?

Reply:
I'm 99% sure we're capitalizing when it's used as a feature name, so it's correct in the first sentence.

Just noticed the second sentence though - I think that should be lowercase:

> A Branch Deployment is a new Dagster deployment